From c4d06b521ac6fc9bfb611646d01fd706516d8274 Mon Sep 17 00:00:00 2001
From: Michael Stapelberg <stapelberg@debian.org>
Date: Thu, 21 Mar 2013 00:37:47 +0100
Subject: [PATCH] systemd: update check to look for /run/systemd/system

The old check (for /sys/fs/cgroup/systemd) is a false-positive on
systems which use standalone logind, but not systemd.

The new one is recommended and used by upstream systemd in their
sd_booted() implementation, see
http://cgit.freedesktop.org/systemd/systemd/tree/src/libsystemd-daemon/sd-daemon.c#n518
---
 autoscripts/postinst-init-nostart  |    2 +-
 autoscripts/postinst-init-restart  |    2 +-
 autoscripts/postinst-init-tmpfiles |    2 +-
 autoscripts/postrm-init            |    2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/autoscripts/postinst-init-nostart b/autoscripts/postinst-init-nostart
index ec6bdbd..6ccb483 100644
--- a/autoscripts/postinst-init-nostart
+++ b/autoscripts/postinst-init-nostart
@@ -1,7 +1,7 @@
 if [ -x "/etc/init.d/#SCRIPT#" ]; then
 	# In case this system is running systemd, we need to ensure that all
 	# necessary tmpfiles (if any) are created before starting.
-	if [ -d /sys/fs/cgroup/systemd ] ; then
+	if [ -d /run/systemd/system ] ; then
 		systemd-tmpfiles --create >/dev/null || true
 	fi
 
diff --git a/autoscripts/postinst-init-restart b/autoscripts/postinst-init-restart
index 862f9cc..d27cc8d 100644
--- a/autoscripts/postinst-init-restart
+++ b/autoscripts/postinst-init-restart
@@ -1,7 +1,7 @@
 if [ -x "/etc/init.d/#SCRIPT#" ]; then
 	# In case this system is running systemd, we need to ensure that all
 	# necessary tmpfiles (if any) are created before starting.
-	if [ -d /sys/fs/cgroup/systemd ] ; then
+	if [ -d /run/systemd/system ] ; then
 		systemd-tmpfiles --create >/dev/null || true
 	fi
 
diff --git a/autoscripts/postinst-init-tmpfiles b/autoscripts/postinst-init-tmpfiles
index 41f738a..e6cb785 100644
--- a/autoscripts/postinst-init-tmpfiles
+++ b/autoscripts/postinst-init-tmpfiles
@@ -1,5 +1,5 @@
 # In case this system is running systemd, we need to ensure that all
 # necessary tmpfiles (if any) are created before starting.
-if [ -d /sys/fs/cgroup/systemd ] ; then
+if [ -d /run/systemd/system ] ; then
 	systemd-tmpfiles --create >/dev/null || true
 fi
diff --git a/autoscripts/postrm-init b/autoscripts/postrm-init
index 61dcfc5..6f5bb09 100644
--- a/autoscripts/postrm-init
+++ b/autoscripts/postrm-init
@@ -5,6 +5,6 @@ fi
 
 # In case this system is running systemd, we make systemd reload the unit files
 # to pick up changes.
-if [ -d /sys/fs/cgroup/systemd ] ; then
+if [ -d /run/systemd/system ] ; then
 	systemctl --system daemon-reload >/dev/null || true
 fi
-- 
1.7.10.4

